tests/sources/test_graphql.py

```diff
@@ -0,0 +1,225 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+"""Tests the GraphQL source class methods"""
```

Comment:
the boil_down_data has some important logic inside, can we add unit tests here as well? 

I will review tests further once we have connector logic ready :)